### ISSUE TITLE
perl-Alien-wxWidgets fails check phase

### ISSUE THREAD

**Issue by Profpatsch** [NixOS/nixpkgs#7099]
```
#   Failed test 'use Alien::wxWidgets;'
#   at t/01_load.t line 6.
#     Tried to use 'Alien::wxWidgets'.
#     Error:  Can't locate Module/Pluggable.pm in @INC (you may need to install the Module::Pluggable module) (@INC contains: /nix/store/j1sism3lmxjkpkxrl4cx378fpcfjf4d9-perl-Alien-wxWidgets-0.65/lib/perl5/site_perl /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/site_perl/5.20.1/x86_64-linux-thread-multi /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/site_perl/5.20.1 /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/site_perl /tmp/nix-build-perl-Alien-wxWidgets-0.65.drv-0/Alien-wxWidgets-0.65/blib/lib /tmp/nix-build-perl-Alien-wxWidgets-0.65.drv-0/Alien-wxWidgets-0.65/blib/arch lib inc /tmp/nix-build-perl-Alien-wxWidgets-0.65.drv-0/Alien-wxWidgets-0.65/_build/lib /nix/store/j1sism3lmxjkpkxrl4cx378fpcfjf4d9-perl-Alien-wxWidgets-0.65/lib/perl5/site_perl /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/site_perl/5.20.1/x86_64-linux-thread-multi /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/site_perl/5.20.1 /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/site_perl _build/lib /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/site_perl/5.20.1/x86_64-linux-thread-multi /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/site_perl/5.20.1 /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/site_perl /nix/store/j1sism3lmxjkpkxrl4cx378fpcfjf4d9-perl-Alien-wxWidgets-0.65/lib/perl5/site_perl /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/site_perl/5.20.1/x86_64-linux-thread-multi /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/site_perl/5.20.1 /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/5.20.1/x86_64-linux-thread-multi /nix/store/g9ybksy400pfn7fncw8dqfnz6m7fdyrk-perl-5.20.1/lib/perl5/5.20.1 .) at /tmp/nix-build-perl-Alien-wxWidgets-0.65.drv-0/Alien-wxWidgets-0.65/blib/lib/Alien/wxWidgets.pm line 43.
# BEGIN failed--compilation aborted at /tmp/nix-build-perl-Alien-wxWidgets-0.65.drv-0/Alien-wxWidgets-0.65/blib/lib/Alien/wxWidgets.pm line 43.
# Compilation failed in require at t/01_load.t line 6.
# BEGIN failed--compilation aborted at t/01_load.t line 6.
# Looks like you failed 1 test of 1.
t/01_load.t .......... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
t/zy_pod_coverage.t .. skipped: Test::Pod::Coverage 1.00 required for testing POD coverage
t/zz_pod.t ........... skipped: Test::Pod 1.00 required for testing POD

Test Summary Report
-------------------
t/01_load.t        (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=3, Tests=1,  0 wallclock secs ( 0.03 usr  0.01 sys +  0.09 cusr  0.01 csys =  0.14 CPU)
Result: FAIL
Failed 1/3 test programs. 1/1 subtests failed.
make: *** [test] Error 255
builder for ‘/nix/store/cvlgaj0x2802kjj0vcbl8gaxh5dn7k8n-perl-Alien-wxWidgets-0.65.drv’ failed with exit code 2
building path(s) ‘/nix/store/kqg81ksi4dbv73l5sy4vnff74qba31vs-perl-ExtUtils-Typemap-1.00’
cannot build derivation ‘/nix/store/5lkrskvc648zqqjj0ha5r2n8p9bkswk1-perl-Wx-0.9923.drv’: 1 dependencies couldn't be built
building path(s) ‘/nix/store/p83lwjnwngd8s4qm632j6f2n6mkxhdrc-perl-XML-Twig-3.44’
cannot build derivation ‘/nix/store/3j88vdjpcb5jwv2yfv2gs13jiccyl9kg-slic3r-1.2.1.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/xj0gwazrp0rn8vasa1h960i0lrj0nhhp-system-path.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/iisx3lw8f9ji44wiqdxy12fsnkp6xflm-nixos-14.12.555.99bfc89.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/iisx3lw8f9ji44wiqdxy12fsnkp6xflm-nixos-14.12.555.99bfc89.drv’ failed
```

**Comment by vcunat** [comment]
[renamed]... causing slic3r dependency failure.

**Comment by Profpatsch** [comment]
Is there a simple way to work around this, since it’s only the check phase?

Sadly, I don’t know any Perl and am to fresh into nix to know where to look. It’s hard to see what is nix output and what is perl output, too.

**Comment by vcunat** [comment]
Certainly, disabling checks for this package should work around it, at least for this particular package (not certain that slic3r will build right after this).

**Comment by vcunat** [comment]
Note: slic3r contains this comment on master
>     broken = true; # requires Perl 5.14

/cc maintainers @the-kenny, @bjornfor.

**Comment by vcunat** [comment]
Another thing, perl-Alien-wxWidgets tests succeed for me on x86_64-linux. Could it be that you've run it as a non-chrooted build? (Or maybe it's just non-determinitsic.)

**Comment by bjornfor** [comment]
Maybe updating slic3r will remove the need for perl-5.14 (which exists no more in nixpkgs master): https://github.com/alexrj/Slic3r/blob/master/.travis.yml (lists perl 5.20 as some of the environments).

**Comment by Profpatsch** [comment]
Oh, I’m sorry, I was not on nixos-unstable.

**Comment by bjornfor** [comment]
perl-Alien-wxWidgets and slic3r are now fixed in master and release-14.12 branches.

@Profpatsch: The error you pasted comes from the fact that newer perl removed the ModulePluggable module. So now it has to be an explicit dependency.
